graal/com.oracle.graal.truffle/src/com/oracle/graal/truffle/phases/InstrumentBranchesPhase.java

```
@@ -0,0 +1,217 @@
+            else {
+                int index = tableCount++;
+                Point p = new Point(index);
+                pointMap.put(key, p);
+                return p;
+            }
+        }
+
+    }
+
+    public static BranchInstrumentation instrumentation = new BranchInstrumentation();
+
+    public static class SourceLocation {
+        private BytecodePosition position;
+
+        public SourceLocation(BytecodePosition position) {
+            this.position = position;
+        }
+
+        public BytecodePosition getPosition() {
+            return position;
+        }
+    }
+
+    public InstrumentBranchesPhase() {
+
+    }
+
+    private void insertCounter(StructuredGraph graph, ResolvedJavaField tableField, JavaConstant tableConstant,
+                    IfNode ifNode, BranchInstrumentation.Point p, boolean isTrue) {
+        assert (tableConstant != null);
+        AbstractBeginNode beginNode = (isTrue) ? ifNode.trueSuccessor() : ifNode.falseSuccessor();
+        ConstantNode table = graph.unique(new ConstantNode(tableConstant, StampFactory.exactNonNull((ResolvedJavaType) tableField.getType())));
+        ConstantNode rawIndex = graph.unique(ConstantNode.forInt(p.getRawIndex(isTrue)));
+        ConstantNode v = graph.unique(ConstantNode.forBoolean(true));
+        StoreIndexedNode store = graph.add(new StoreIndexedNode(table, rawIndex, JavaKind.Boolean, v));
+
+        graph.addAfterFixed(beginNode, store);
+    }
+
+    public static void addNodeSourceLocation(Node node, BytecodePosition pos) {
+        node.setNodeContext(new SourceLocation(pos));
+    }
+
+    @Override
+    protected void run(StructuredGraph graph, HighTierContext context) {
+        if (METHOD_REGEX_FILTER.matcher(graph.method().getName()).matches()) {
+            Field javaField = null;
+            try {
+                javaField = InstrumentBranchesPhase.class.getField("TABLE");
+            } catch (Exception e) {
+                return;
```

Never swallow exceptions. Its better to wrap it in a RuntimeException and rethrow.